// usebootstrap/dist/runtime/plugins/seometa.js
import { joinURL } from "ufo";
import { useDynamicRouteParam } from "../composables/extend/dynamicRoute/useDynamicRouteParam.js";
import { defineNuxtPlugin, useRoute } from "#app";
import { useSeoMeta, useSiteConfig } from "#imports";
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.hook("page:loading:start", () => {
    const siteConfig = useSiteConfig();
    const siteUrl = siteConfig?.url;
    const siteName = siteConfig?.name;
    const createImage = () => {
      if (siteUrl) {
        if (siteConfig?.image) {
          return joinURL(siteUrl, siteConfig?.image);
        }
      }
      return void 0;
    };
    const lang = useDynamicRouteParam("lang");
    const route = useRoute();
    useSeoMeta({
      title: () => route.meta[`title:${lang.value}`] || route.meta.title,
      description: () => route.meta[`description:${lang.value}`] || route.meta.description,
      ogTitle: () => route.meta[`title:${lang.value}`] || route.meta.title,
      ogDescription: () => route.meta[`description:${lang.value}`] || route.meta.description,
      ogSiteName: siteName,
      ogUrl: siteUrl,
      ogLocale: lang.value,
      ogImage: createImage(),
      twitterDescription: () => route.meta[`description:${lang.value}`] || route.meta.description,
      twitterCard: () => createImage() ? "summary_large_image" : void 0,
      twitterTitle: () => route.meta[`title:${lang.value}`] || route.meta.title,
      twitterImage: createImage()
    });
  });
});
